Add HomeConnect services documentation (#20098)
home-assistant/home-assistant.github.io: source/_integrations/home_connect.markdown

{% include integrations/config_flow.md %}

The integration configuration will ask for the *Client ID* and *Client Secret* created above. See [Application Credentials](/integrations/application_credentials) for more details.

## Services

The Home Connect integration makes various services available.
Available services: `set_option_active`, `set_option_selected`, `pause_program`, `resume_program`, `select_program`, `start_program` and `change_setting`

### Service `home_connect.set_option_active`

Sets an option for the active program.

| Service data attribute    | Optional | Description                                      |
|---------------------------|----------|--------------------------------------------------|

> | `entity_id` | no | Id of an entity associated with the home appliance. |

Reviewer: ```suggestion
| `device_id` | no | Id of a device associated with the home appliance. |
```

| `key` | no | Key of the option. |
| `value` | no | Value of the option. |
| `unit` | yes | Unit for the option. |

### Service `home_connect.set_option_selected`

Sets an option for the selected program.

| Service data attribute    | Optional | Description                                      |
|---------------------------|----------|--------------------------------------------------|

> | `entity_id` | no | Id of an entity associated with the home appliance. |

Reviewer: ```suggestion
| `device_id` | no | Id of a device associated with the home appliance. |
```

| `key` | no | Key of the option. |
| `value` | no | Value of the option. |
| `unit` | yes | Unit for the option. |

### Service `home_connect.pause_program`

Pauses the current running program.

| Service data attribute    | Optional | Description                                      |
|---------------------------|----------|--------------------------------------------------|

> | `entity_id` | no | Id of an entity associated with the home appliance. |

Reviewer: ```suggestion
| `device_id` | no | Id of a device associated with the home appliance. |
```

### Service `home_connect.resume_program`

Resumes a paused program.

| Service data attribute    | Optional | Description                                      |
|---------------------------|----------|--------------------------------------------------|

> | `entity_id` | no | Id of an entity associated with the home appliance. |

Reviewer: ```suggestion
| `device_id` | no | Id of a device associated with the home appliance. |
```

### Service `home_connect.select_program`

Selects a program without starting it.

| Service data attribute    | Optional | Description                                      |
|---------------------------|----------|--------------------------------------------------|

> | `entity_id` | no | Id of an entity associated with the home appliance. |

Reviewer: ```suggestion
| `device_id` | no | Id of a device associated with the home appliance. |
```

| `program` | no | Program to select. |
| `key` | yes | Key of the option. |
| `value` | yes | Value of the option. |
| `unit` | yes | Unit for the option. |

### Service `home_connect.start_program`

Selects a program and starts it.

| Service data attribute    | Optional | Description                                      |
|---------------------------|----------|--------------------------------------------------|

> | `entity_id` | no | Id of an entity associated with the home appliance. |

Reviewer: ```suggestion
| `device_id` | no | Id of a device associated with the home appliance. |
```

| `program` | no | Program to select. |
| `key` | yes | Key of the option. |
| `value` | yes | Value of the option. |
| `unit` | yes | Unit for the option. |

### Service `home_connect.change_setting`

Changes a setting.

| Service data attribute    | Optional | Description                                      |
|---------------------------|----------|--------------------------------------------------|

> | `entity_id` | no | Id of an entity associated with the home appliance. |

Reviewer: ```suggestion
| `device_id` | no | Id of a device associated with the home appliance. |
```
